### PR TITLE
Added windows support for hub and node

### PR DESCRIPTION
**Pull request by jollyrojer**
Please, set tag 2.0-43p

### CODE REVIEW

**Comment by jollyrojer** [comment]
retest this please

**Comment by jollyrojer** [comment]
retest this please

**Comment by jollyrojer** [comment]
retest this please

File: test/test_remote_webdriver.py [hunk 1]
```diff
@@ -6,50 +6,65 @@
...
-                desired_capabilities=capability)
-            driver.get(console)
-            assert driver.title == "Grid Console"
-        finally:
-            if driver:
-                driver.quit()
+  def console_base(self, instance, capability):
+    console = instance.returnValues['endpoints.console-url']
+    remote = instance.returnValues['endpoints.remote-url']
+    time.sleep(120)  # let node establish connection with hub
```
Comment (dmakhno) [inline test/test_remote_webdriver.py]: this is too much.

better
```python
@retry(30,4,1, WebDriverException)
def get_driver():
  return webdriver.Remote(
        command_executor=str(remote),
        desired_capabilities=capability)
...
try:
  driver = get_driver()
```

File: test/test_remote_webdriver.py [hunk 2]
```diff
@@ -6,50 +6,65 @@
...
-    @values({"endpoints.console-url": "console", "endpoints.remote-url": "remote"})
-    def test_firefox(self, instance, console, remote):
-        self.console_base(DesiredCapabilities.FIREFOX, console, remote)
+  @instance(byApplication='Linux selenium-hub')
+  def test_lin_firefox(self, instance):
+        self.console_base(instance, DesiredCapabilities.FIREFOX)
+
+@applications([{
+  "name": 'Windows selenium-hub',
+  "file": os.path.realpath(os.path.join(os.path.dirname(__file__), '../component-selenium-grid.yml')),
```
Comment (dmakhno) [inline test/test_remote_webdriver.py]: dry

File: component-selenium-grid.yml [hunk 3]
```diff
@@ -1,104 +1,264 @@
...
-     console-url: bind(selenium-grid#result.console-url)
-     remote-url: bind(selenium-grid#result.remote-url)
+     console-url: bind(selenium-hub#result.console-url)
+     remote-url: bind(selenium-hub#result.remote-url)
     hosts:
-      hub: bind(selenium-grid#result.hub-host)
-      nodes: bind(selenium-grid#result.node-hosts)
-    actions:
-      scale: bind(selenium-grid#actions.scale)
-      re-start: bind(selenium-grid#actions.re-start)
```
Comment (dmakhno) [inline component-selenium-grid.yml]: selenium might have memory leaks, and it is recommended to restart cluster from time to time.

File: component-selenium-grid.yml [hunk 4]
```diff
@@ -1,104 +1,264 @@
...
-          selenium-version:
-            type: configuration(string)
-            name: Selenium Version
-          node-count:
-            type: configuration(string)
-            name: Count of Additional Nodes
-          recipes:
-            type: configuration(string)
-            name: Recipes Url
+          hub-image:
```
Comment (dmakhno) [inline component-selenium-grid.yml]: name is missing

File: component-selenium-grid.yml [hunk 5]
```diff
@@ -1,104 +1,264 @@
...
+              "Win2008r2 (us-west-1)": { ami: "us-west-1/ami-2b2dd76f", user: "Administrator", type: "windows", hw: "m3.large", issudo: false, prefix: "Selenium-hub"  }
+              "CentOS 7.1 (us-east-1)": { ami: "us-east-1/ami-96a818fe", user: "centos", type: "linux", hw: "m3.medium", issudo: true, prefix: "Selenium-hub" }
+              "CentOS 7.1 (us-west-1)": { ami: "us-west-1/ami-6bcfc42e", user: "centos", type: "linux", hw: "m3.medium", issudo: true, prefix: "Selenium-hub" }
+          selenium-version: { type: configuration(string), name: Selenium Version }
+          ie-driver-version: { type: configuration(string), name: IE Driver Version }
+          chrome-driver-version: { type: configuration(string), name: Chrome Driver Version }
+          recipes: { type: configuration(string), name: Recipes Url }
+          java-win-url: { type: configuration(string), name: "Windows Java installer URL" }
+          windows-user-password: { type: configuration(string), name: "Windows Administrator user password" }
+          mirror: { type: configuration(string), name: "Cygwin installation binary URL" }
```
Comment (dmakhno) [inline component-selenium-grid.yml]: strange pin name

**Comment by jollyrojer** [comment]
retest this please

File: component-selenium-grid.yml (hunk 5 above)
Comment (jollyrojer) [inline component-selenium-grid.yml]: We use it everywhere. You'd prefer cygwin-url? Does it make any difference as name exposed like "Cygwin installation binary URL"?

**Comment by jollyrojer** [comment]
retest this please

**Comment by jollyrojer** [comment]
retest this please

**Comment by jollyrojer** [comment]
retest this please

File: test/test_remote_webdriver.py [hunk 6]
```diff
@@ -3,53 +3,76 @@
...
+        command_executor=str(remote),
+        desired_capabilities=capability)
+
+    driver = None
+    try:
+      driver = get_driver()
+      driver.get(console)
+      assert driver.title == "Grid Console"
+    finally:
+      if driver:
```
Comment (dmakhno) [inline test/test_remote_webdriver.py]: try:
  driver.quit()
except:
  pass

File: component-selenium-grid.yml [hunk 7]
```diff
@@ -107,151 +266,275 @@ application:
...
-                  roles: ["{$.role}"]
-                  runList: ${run-list}
-                  version: 11.18.6
-                  isSolo: true
-                  recipeUrl: "{$.recipes}"
-                  jattrs:
-                    selenium:
-                      server:
-                        cleanUpCycle: 20000
-                        timeout: 120000
```
Comment (dmakhno) [inline component-selenium-grid.yml]: is there way to configure these timeouts now?

File: component-selenium-grid.yml (hunk 5 above)
Comment (dmakhno) [inline component-selenium-grid.yml]: `cygwin-url` seems more proper name.
Display name should correspond to pin name at least somehow.

File: component-selenium-grid.yml (hunk 7 above)
Comment (jollyrojer) [inline component-selenium-grid.yml]: Yes you can. I've added an example to manifest.

**Pull request merged**